cuqi/distribution/_modifiedhalfnormal.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+import scipy.stats as sps
+import scipy.special as special
+from cuqi.distribution import Distribution
+from cuqi.utilities import force_ndarray
+
+class ModifiedHalfNormal(Distribution):
+    """
+    Represents a modified half-normal (MHN) distribution, a three-parameter family of distributions generalizing the Gamma distribution.
+    The distribution is continuous with pdf 
+    f(x; alpha, beta, gamma) propto x^(alpha-1) * exp(-beta * x^2 + gamma * x)
+
+    The MHN generalizes the half-normal distribution, because
+    f(x; 1, beta, 0) propto exp(-beta * x^2)
+
+    The MHN generalizes the gamma distribution because
+    f(x; alpha, 0, -gamma) propto x^(alpha-1) * exp(- gamma * x)
+
+    Reference:
+    [1] Sun, et al. "The Modified-Half-Normal distribution: Properties and an efficient sampling scheme." Communications in Statistics-Theory and Methods
+    
+    Parameters
+    ----------
+    alpha : float
+        The polynomial exponent parameter of the MHN distribution. Must be positive.
+
+    beta : float
+        The quadratic exponential parameter of the MHN distribution. Must be positive.
+
+    gamma : float
+        The linear exponential parameter of the MHN distribution.
+
+    """
+    def __init__(self, alpha=None, beta=None, gamma=None, is_symmetric=False, **kwargs):
+        # Init from abstract distribution class
+        super().__init__(is_symmetric=is_symmetric, **kwargs) 
+
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma

```

Comment:
Also. If you could add getters and setters for these 3 parameters with force_ndarray in them, then the unit tests that are skipped in test_distributions_shape should hopefully pass.

Reply:
I have added the requested getters and setters, with the force_ndarray, yet the tests still fail. With the Modified Half-Normal being a generalized Gamma distribution, I suppose the issues are the same as why these tests are skipped for the Gamma distribution and other related distributions. So if the issue is the same, shouldn't they all be fixed in one go, perhaps not as part of this pull request?

Reply:
To me it looks like the logd returns a vector. We generally assume that logd should return a scalar value. E.g. for multivariate Normal we sum the logd vector (see https://github.com/CUQI-DTU/CUQIpy/blob/b3c53f2ac21c807186dc35053421615c1cb3f205/cuqi/distribution/_normal.py#L40) Hopefully that helps?

Reply:
Thanks for the suggestion. I have fixed it. Another issue with the failed test is that is uses np.allclose(a, b), which returns False if both a and b are NaN, using np.allclose(a, b, equal_nan = True) will fix this issue. This happens when logpdf is evaluated outside its domain. With these changes, all tests now pass.

Reply:
Perfect. Thank you!